samples/net/zperf/sample.yaml

```diff
@@ -10,6 +10,10 @@ sample:
 tests:
   sample.net.zperf:
     platform_allow: qemu_x86
+  sample.net.zperf_no_shell:
+    extra_configs:
+      - CONFIG_NET_SHELL=n
```

Comment:
There are mixed dependencies about the shell.
Here we set `CONFIG_NET_SHELL` but in various settings we depend on `CONFIG_SHELL` which imho looks wrong. As this is network library we should depend on `CONFIG_NET_SHELL` only (it will depend on `CONFIG_SHELL`)